### PR TITLE
Create Default Provider and Hooks for `@realm/react`

### LINKED ISSUE #5292: Export a default Realm context from `@realm/react`
### Problem

It's annoying for users to have to create a Realm context (calling `createRealmContext`) before accessing the `useRealm`, `useQuery` and `useObject` hooks:

```tsx
import { createRealmContext } from "@realm/react";
const { RealmProvider, useRealm } = createRealmContext();
// use the useRealm hook
```

### Solution

Instead we should create a default Realm context and export its members as named exports, making the average use of the library much simpler:

```tsx
import { RealmProvider, useRealm } from "@realm/react";
// use the useRealm hook
```

We should make sure to document the advanced use-case of working with multiple Realm's (calling `createRealmContext`) too.

### Alternatives

_No response_

### How important is this improvement for you?

I would like to have it but have a workaround

### Feature would mainly be used with

Atlas Device Sync

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
* Creates a default context and exports the results.
* Provide the tsdoc directly on the function exports.
* This makes the single realm use case much simpler to start with.  For multi realm usage, the `createRealmContext` is still available.

This closes #5292

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated

### CODE REVIEW

File: packages/realm-react/src/index.tsx [hunk 1]
```diff
@@ -24,79 +24,9 @@ import { createUseObject } from "./useObject";
...
-  /**
-   * Returns the instance of the {@link Realm} configured by `createRealmContext`
-   *
-   * @example
-   * ```
-   * const realm = useRealm();
-   * ```
-   *
-   * @returns a realm instance
-   */
```
Comment (kraenhansen) [inline packages/realm-react/src/index.tsx:99]: I don't think this should be removed. It should still be there for people that need multiple nested Realm providers.

File: packages/realm-react/src/index.tsx [hunk 2]
```diff
@@ -141,6 +71,87 @@ export const createRealmContext: (realmConfig?: Realm.Configuration) => RealmCon
...
+ * }
+ * ```
+ * @param props - The {@link Realm.Configuration} for this Realm defaults to
+ * the config passed to `createRealmProvider`, but individual config keys can
+ * be overridden when creating a `<RealmProvider>` by passing them as props.
+ * For example, to override the `path` config value, use a prop named `path`
+ * e.g., `path="newPath.realm"`
+ * an attribute of the same key.
+ */
+const RealmProvider = defaultContext.RealmProvider;
```
Comment (kraenhansen) [inline packages/realm-react/src/index.tsx:102]: Might as well `export` right here?
```suggestion
export const RealmProvider = defaultContext.RealmProvider;
```

**Review by kraenhansen — APPROVED** [review]
I only have a few suggestions.

File: packages/realm-react/src/index.tsx (hunk 1 above)
Comment (takameyer) [reply packages/realm-react/src/index.tsx:99]: I'm curious how we can sanely remove this duplication and also have better documentation.

File: packages/realm-react/src/index.tsx (hunk 2 above)
Comment (takameyer) [reply packages/realm-react/src/index.tsx:102]: Sure, no reason not to.

**Pull request merged**